### PR TITLE
Canceling release resource if helm 2 release is not deleted

### PR DESCRIPTION
**Pull request by tomahawk28**
Following https://github.com/giantswarm/giantswarm/issues/10973#issuecomment-634473564 

chart-operator is creating helm 3 release even before app-operator starts cordon all charts and start the migration. 

This patch would prevent that early behavior.

### CODE REVIEW

File: service/controller/chart/resource/release/resource.go [hunk 1]
```diff
@@ -89,6 +103,20 @@ func (r *Resource) Name() string {
 	return Name
 }
 
+func (r *Resource) findHelmV2ConfigMaps(ctx context.Context, releaseName string) (bool, error) {
+	lo := metav1.ListOptions{
+		LabelSelector: fmt.Sprintf("%s=%s,%s=%s", "NAME", releaseName, "OWNER", "TILLER"),
+	}
+
+	// Check whether it keep helm2 release configMaps
```
Comment (rossf7) [inline service/controller/chart/resource/release/resource.go:111]:
```suggestion
	// Check whether there are still helm2 release configmaps.
```

File: service/controller/chart/resource/release/current.go [hunk 2]
```diff
@@ -30,6 +30,18 @@ func (r *Resource) GetCurrentState(ctx context.Context, obj interface{}) (interf
 		return nil, nil
 	}
 
+	hasConfigmap, err := r.findHelmV2ConfigMaps(ctx, key.ReleaseName(cr))
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	if hasConfigmap {
+		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("release %#q has been not migrated from helm 2", key.ReleaseName(cr)))
```
Comment (rossf7) [inline service/controller/chart/resource/release/current.go:39]:
```suggestion
		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("release %#q has not been migrated from helm 2", key.ReleaseName(cr)))
```

File: service/controller/chart/resource/release/create_test.go [hunk 3]
```diff
@@ -75,6 +75,8 @@ func Test_Resource_Release_newCreate(t *testing.T) {
 			HelmClient: helmclienttest.New(helmclienttest.Config{}),
 			K8sClient:  k8sfake.NewSimpleClientset(),
 			Logger:     microloggertest.New(),
+
+			TillerNamespace: "giantswarm",
```
Comment (rossf7) [inline service/controller/chart/resource/release/create_test.go:79]: Not for now but I'm wondering if we should drop the unit tests apart from the desired state.

I think we already did that in app-operator. I'd rather extend our existing integration tests or add new ones. WDYT?

File: service/controller/chart/resource/release/current.go (hunk 2 above)
Comment (tomahawk28) [reply service/controller/chart/resource/release/current.go:39]: Thanks for helping 🇬🇧 :)

File: service/controller/chart/resource/release/create_test.go (hunk 3 above)
Comment (tomahawk28) [reply service/controller/chart/resource/release/create_test.go:79]: Good call, I would delete them in next PR.

**Pull request merged**